numpy/core/src/umath/ufunc_object.c

```diff
@@ -4360,13 +4358,17 @@ ufunc_generic_call(PyUFuncObject *ufunc, PyObject *args, PyObject *kwds)
         int j = ufunc->nin+i;
         PyObject *wrap = wraparr[i];
 
-        if (wrap != NULL) {
+        if (wrap == NULL) {
+            /* default behavior */
+            retobj[i] = PyArray_Return(mps[j]);
+        }
+        else if (wrap == Py_None) {
+            Py_DECREF(wrap);
+            retobj[i] = (PyObject *)mps[j];
+        }
+        else {
+            PyObject *res;
             PyObject *args_tup;
-            if (wrap == Py_None) {
-                Py_DECREF(wrap);
-                retobj[i] = (PyObject *)mps[j];
-                continue;
-            }
 
             /* Call the method with appropriate context */
             args_tup = _get_wrap_prepare_args(full_args);
```

Comment:
Not quite this PR, but while you're at it: there would seem to be no reason to call this every time in the loop; maybe preset `args_tup` to `NULL` on top, and here only set it if `NULL`? I.e.,
```
if (args_tup == NULL) {
    args_tup = _get_wrap_prepare_args(full_args);
    if (args_tup == NULL) {
        goto fail;
    }
}
```

Reply:
I also considered just caching this on the `full_args` object. Something for another PR, I think.

Reply:
OK, that's fine.